app/controllers/notifications_controller.rb

```diff
@@ -66,4 +67,12 @@ def if_not_signed_in
         end
       end
     end
+
+    # Stops here if the signed in user is not the owner of
+    # the notification set by :set_notification.
+    def ensure_user_owns_notification!
+      unless @notification.user == current_user
```

Comment:
Personally, there seems to be a lot of indirection with the assignment of `@notification`.  It's being assigned in `set_notification`, then validated here, and finally actually used in only one place: the destroy action.  It's ok right now because this controller is still pretty simple, but I've seen as Rails apps grow, this type of juggling of instance variables between the before actions and the actual actions can get confusing.

What if we were to combine this new before action with the existing `set_notification` one so that the assignment and validation can happen in a single place?  Just my two cents! 👍